### PR TITLE
Show proper error message when user is not signed up for the customizer or is trying to publish an empty workspace

### PR DESCRIPTION
**Pull request by pboyer**
### Purpose

When a user is not signed up for the customizer they get a 401 HTTP status code from the server. This was being swallowed up and not displayed properly. Now, by responding with a more descriptive response type, `IReachHttpResponse`, we can easily get at this info.

I also cleaned up the code to use `async`, `await` - which makes the code a heck of a lot more readable I think.

While I was at it, I also fixed MAGN-8875, which is annotated below. This PR requires some minor changes to Reach.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@ramramps

### CODE REVIEW

File: src/DynamoPublish/Models/PublishModel.cs [hunk 1]
```diff
@@ -276,86 +279,87 @@ internal void Authenticate()
...
+                return;
             }
 
-            if (authenticationProvider == null)
+            if (workspace.Nodes.Count() == 0)
             {
-                Error = UploadErrorType.AuthProviderNotFound;
-                return Resources.FailedMessage;
+                Error = UploadErrorType.EmptyWorkspace;
+                return;
```
Comment (pboyer) [inline src/DynamoPublish/Models/PublishModel.cs]: This is the change for http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8875.

File: src/DynamoPublish/Models/PublishModel.cs [hunk 2]
```diff
@@ -276,86 +279,87 @@ internal void Authenticate()
...
 
-            string result;
             try
             {
-                result = reachClient.Send(
-                    workspace,
-                    dependencies.CustomNodeWorkspaces,
-                    workspaceProperties);
-                InvalidNodeNames = null;
+                result = await Task.Run(() => this.Send(workspace, workspaceProperties));
```
Comment (ramramps) [inline src/DynamoPublish/Models/PublishModel.cs]: You can do this with  await Task.Run().ContinueWith - this gives a more controlled way to access the result message. 
 Task.Run(() => this.Send(workspace, workspaceProperties).ContinueWith(result => { 
   //check the status code and content here.
});

**Comment by ramramps** [comment]
One suggestion and then LGTM.

**Comment by pboyer** [comment]
@ramramps I think you may have missed the whole purpose of `await` as used here! 

It provides syntactic sugar around what you suggested along with improved exception handling. The code reads so linearly that you wouldn't even know it was doing that task async. Note that I can simply put a try catch around the `result = await Task.Run(...)` and I can catch the exception right there.

**Pull request merged**